include/deal.II/differentiation/ad/ad_number_traits.h

```diff
@@ -752,9 +774,9 @@ namespace Differentiation
       ScalarType,
       ADNumberTypeCode,
       typename std::enable_if<
-        std::is_floating_point<ScalarType>::value ||
+        std::is_arithmetic<ScalarType>::value ||
```

Comment:
At some point there was a compilation error if one of these (either here or below) wasn't set to `is_arithmetic` rather than `is_floating_point`. However, I've changed all instances of `is_arithmetic` to `is_floating_point` in this header and things now work fine. I'll revert this change everywhere.